src/nvim/undo.c

```diff
@@ -2321,7 +2322,8 @@ static void u_undoredo(int undo)
   if (undo)
     /* We are below the previous undo.  However, to make ":earlier 1s"
      * work we compute this as being just above the just undone change. */
-    --curbuf->b_u_seq_cur;
+    curbuf->b_u_seq_cur =  curhead->uh_next.ptr ?
+        curhead->uh_next.ptr->uh_seq : 0;
```

Comment:
I read the comment as being just fine, but the calculation as being incorrect -- i.e. in this error case, instead of computing the sequence number as being just above the just undone change, it computes it as being to the side (on a different branch).

I could replace `"just above the just undone change"` to something like `"just above the just undone change in the tree"` if you think that's clearer.